### ISSUE TITLE
/group endpoint leaks RPC endpoints

### ISSUE THREAD

**Issue by hsanjuan** [drand/drand#379]
The group endpoint should probably give information to verify the randomness, but I don't think it should be leaking the RPC endpoints of everyone: these endpoints are not for public use. At best people will get confused and try to use them as API endpoints. At worst you are giving a list of all the potential attack targets to everyone.

It seems including the public key also reveals the endpoint, which means the public key should not be tied to a drand node location.

Can we tigthen down this endpoint so that it provides a bare minimum of information?

**Comment by nikkolasg** [comment]
Yes i think on a general level we need to make a "private group" and a "public group".

It may not be the best of design but simply have the following design would solve our problems in a straightforward (read "fast") way:
```go
func (g *Group) PublicInfo() *PublicGroup { ... }

type PublicGroup struct {
   Key []byte
   Period uint32
   GenesisTime int64
 }

// Hash returns a canonical representation of a given drand chain, regardless of the network composition
func (p *PublicGroup) Hash() []byte { ... }
```

**Comment by alanshaw** [comment]
The gossip relay topic could then be based on this 😀

**Comment by nikkolasg** [comment]
Fixed in #396
